sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -150,8 +150,66 @@ static const struct hda_dai_widget_dma_ops hda_dma_ops = {
 	.release_hext_stream = hda_release_hext_stream,
 };
 
+static unsigned int get_max_bps(struct snd_sof_dev *sdev, struct snd_soc_dai *codec_dai,
+				struct snd_pcm_substream *substream)
+{
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		return codec_dai->driver->playback.sig_bits;
+
+	return codec_dai->driver->capture.sig_bits;
+}
+
+static int hda_codec_dai_params(struct snd_sof_dev *sdev, struct snd_soc_dai *codec_dai,
+				struct hdac_ext_stream *hext_stream,
+				struct snd_pcm_substream *substream)
+{
+	struct hdac_ext_link *hlink;
+	struct hdac_bus *bus;
+
+	bus = sof_to_bus(sdev);
+	hlink = snd_hdac_ext_bus_get_hlink_by_name(bus, codec_dai->component->name);
+	if (!hlink)
+		return -EINVAL;
+
+	if (hext_stream->hstream.direction == SNDRV_PCM_STREAM_PLAYBACK) {
+		int stream_tag = hdac_stream(hext_stream)->stream_tag;
+
+		snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag);
+	}
```

Comment:
this is a bit awkward. All we need here is something that return the hlink. The rest to set the stream_id is going to be completely codec-agnostic.

In this case we use the codec_dai only to get the hlink information. We could do this differently by checking the name of the CPU DAI.